server/controllers/userController.js

```diff
@@ -13,6 +50,8 @@ const authUser = asyncHandler(async (req, res) => {
     const user = await User.findOne({ email })
 
     if (user && (await user.matchPassword(password))) {
+        onlineUsers[user.id] = true
```

Comment:
Technically a user is online when they have the application open, as opposed to when they are logged in. So instead, I would send events from the front-end React app when the user opens the application and when they close the application, to know if they are currently online